nevergrad/functions/gym/gym_multi.py

```diff
@@ -0,0 +1,924 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+import numpy as np
+import os
+import typing as tp
+import gym
+
+compiler_gym_present = True
+try:
+    import compiler_gym  # pylint: disable=unused-import
+    from compiler_gym.envs import CompilerEnv
+except ImportError:
+    compiler_gym_present = False
+
+import nevergrad as ng
+
+if os.name != "nt":
+    import gym_anm  # pylint: disable=unused-import
+from nevergrad.parametrization import parameter
+from ..base import ExperimentFunction
+
+
+GYM_ENV_NAMES = []
+
+for e in gym.envs.registry.all():
+    try:
+        assert "Kelly" not in str(e.id)
+        assert "llvm" not in str(e.id)
+        env = gym.make(e.id)
+        env.reset()
+        env.step(env.action_space.sample())
+        a1 = np.asarray(env.action_space.sample())
+        a2 = np.asarray(env.action_space.sample())
+        a3 = np.asarray(env.action_space.sample())
+        a1 = a1 + a2 + a3
+        if hasattr(a1, "size"):
+            try:
+                assert a1.size < 15000  # type: ignore
+            except:
+                assert a1.size() < 15000  # type: ignore
+        GYM_ENV_NAMES.append(e.id)
+    except:
+        pass
```

Comment:
please no inline code, move this to a function

Reply:
done.